python/cudf/cudf/tests/test_options.py

```diff
@@ -45,33 +47,96 @@ def validator(x):
     del cudf.options._OPTIONS["even_option"]
 
 
-def test_option_get_set(odd_option):
-    assert cudf.get_option("odd_option") == 1
-    cudf.set_option("odd_option", 101)
-    assert cudf.get_option("odd_option") == 101
-
-
-def test_option_set_invalid(odd_option):
-    with pytest.raises(ValueError, match="Invalid option value 0"):
-        cudf.set_option("odd_option", 0)
-
+@pytest.mark.usefixtures("odd_option", "even_option")
+class TestCleanOptions:
+    def test_option_get_set(odd_option):
+        assert cudf.get_option("odd_option") == 1
+        cudf.set_option("odd_option", 101)
+        assert cudf.get_option("odd_option") == 101
+
+    def test_option_set_invalid(odd_option):
+        with pytest.raises(ValueError, match="Invalid option value 0"):
+            cudf.set_option("odd_option", 0)
+
+    def test_option_description(odd_option):
+        s = StringIO()
+        with redirect_stdout(s):
+            cudf.describe_option("odd_option")
+        s.seek(0)
+        expected = (
+            "odd_option:\n\tAn odd option.\n\t[Default: 1] [Current: 1]\n"
+        )
+        assert expected == s.read()
+
+    def test_option_description_all(odd_option, even_option):
+        s = StringIO()
+        with redirect_stdout(s):
+            cudf.describe_option()
+        s.seek(0)
+        expected = (
+            "odd_option:\n\tAn odd option.\n\t[Default: 1] [Current: 1]\n"
+            "even_option:\n\tAn even option.\n\t[Default: 0] [Current: 0]\n"
+        )
+        assert expected == s.read()
+
+
+def test_empty_option_context():
+    prev_setting = cudf.get_option("default_integer_bitwidth")
+    set_default_integer_bitwidth = random.choice(
```

Comment:
Don’t randomly select. Write a deterministic test. You could parameterize over multiple values or set the value multiple times in the same test to ensure that at least one of them is not the default.

Reply:
And maybe we only need to test one option here, not pandas compat and bitwidth and spilling and etc.

Reply:
Good idea, Simplified it with parametrization.